app/assets/javascripts/discourse-common/addon/lib/dom-from-string.js

```diff
@@ -0,0 +1,6 @@
+export default function domFromString(string) {
+  const template = document.createElement("template");
+  string = string.trim();
+  template.innerHTML = string;
+  return template.content.firstChild;
```

Comment:
If this string includes some kind of tag like `<img>` or `<script>`, this will trigger a network request, even if the element is never added to the DOM.

I wonder if we should be using a 'detached' document, like we do when generating/decorating `PostCooked`?

https://github.com/discourse/discourse/blob/0719531bd37530341616745d39b3d155c9468358/app/assets/javascripts/discourse/app/widgets/post-cooked.js#L27-L31

(I realise this issue exists before this PR, just thought it might be a nice thing to clean up)

Reply:
@jjaffeux just explained to me about the `<template>` element - TIL. Looks like this is not an issue - ignore me

Reply:
Seen with david in chat, `createElement("template")` is fine and not impacted by this. This is certainly a behavior to be aware of though when using classic createElement("div") for example.